src/corelibs/U2Core/src/tasks/ExternalToolRunTask.cpp

```diff
@@ -60,11 +64,17 @@ ExternalToolRunTask::ExternalToolRunTask(const QString& _toolId, const QStringLi
       helper(nullptr),
       listener(nullptr),
       parseOutputFile(parseOutputFile) {
-    CHECK_EXT(AppContext::getExternalToolRegistry()->getById(toolId) != nullptr, stateInfo.setError(tr("External tool is absent")), );
-    QString toolPath = AppContext::getExternalToolRegistry()->getById(toolId)->getPath();
-    CHECK_EXT(QFile::exists(toolPath), stateInfo.setError(tr("External tool '%1' doesn't exist").arg(toolPath)), )
-
-    toolName = AppContext::getExternalToolRegistry()->getToolNameById(toolId);
+    ExternalTool* tool = AppContext::getExternalToolRegistry()->getById(toolId);
+    CHECK_EXT(tool != nullptr, stateInfo.setError(tr("External tool is absent")), );
+    CHECK_EXT(QFile::exists(tool->getPath()), stateInfo.setError(tr("External tool '%1' doesn't exist").arg(tool->getPath())), )
+    
+    const QString error = tool->checkPaths(arguments);
```

Comment:
You check tool arguments only here. What about the tool path itself?

Reply:
We check everything in this method.